syntax = "proto3";

package model;

import "common.proto";

service SimilarityService { 
	rpc GetRankedFaceVector(FeatureRankingRequest) returns (FeatureRankingResponse) {}
	rpc GetRankedCarVector(FeatureRankingRequest) returns (FeatureRankingResponse) {}
	rpc GetRankedVector(FeatureRankingRequest) returns (FeatureRankingResponse) {}
}

message FeatureRankingRequest {
	int64 ReqId = 1;
	// This is the image data.
	Image Image = 2;
	// This is the interested areas.
	repeated Cutboard InterestedAreas = 3;
	repeated FeatureVector Candidates = 4;
	// Limit limits the count of results returned.
	// If zero, returns all.
	int32 Limit = 5;
}

message FeatureRankingResponse {
	int64 ReqId = 1;
	repeated int64 Ids = 2;
	repeated float Scores = 3;
}
